FordLabs/peoplemover
# Upgrade husky and add lint-staged pre-commit hooks

## Issue
Resolves N/A

## What was done
- [x] Upgraded [Husky](https://typicode.github.io/husky/#/) to version 7 which uses a .husky folder instead of a husky object in the package.json
- [x] Moved husky's installation scripts from the root to the UI package.json and deleted the root package.json
- [x] Added [lint-staged](https://github.com/okonet/lint-staged) to handle running certain scripts against different files.

## How to test
1. Run `npm i` in the ui folder.
2. Change something in any tsx file.
3. Make a mock commit (but don't push) and ensure that the lint-staged pre-commit hook runs. It should run all unit tests related to the change you made, and eslint those files.
